### DISCUSSION TITLE
SSR and lazyCompilation

### DISCUSSION THREAD

**Discussion by meskill** [webpack/webpack#15698] [Q&A]
Currently, we heavily use the SSR by building both client & server using webpack's MultiCompiler. In order to get the required scripts and styles from the SSR we read the client's compilation stats.json and use libs such as [loadable components](https://github.com/vercel/next.js/blob/7c6052a084dca523b4f60f8132e67daecd587407/packages/next/server/dev/on-demand-entry-handler.ts#L0-L1)

It will be great to use such a compiler option as `lazyCompilation` that may bring performance improvements. But, it seems it currently doesn't support the SSR well as we cannot gather info about used chunks on the page as they are generated dynamically by webpack and it leads to page flashes due to empty css on page loading.

Does someone has any working example for the SSR and lazyImport or have any advice on how to use both in development without losing SSR features?

**Comment by vankop** [comment]
`lazyCompilation` is only for development..

**Reply by meskill** [reply]
Sure, why would you think I've asked about something else? 😃

**Reply by vankop** [reply]
hm.. Does hmr work with ssr?

**Reply by meskill** [reply]
We use hmr only on the client-side and build the entire app for the server in order to make ssr works. I'm trying to improve overall performance by improving client build time with `lazyCompilation`, but I have issues to gather client info on server with that option

**Reply by vankop** [reply]
`lazyCompilation` works only with hmr ( it is some kind part of hmr )

> We use hmr only on the client-side and build the entire app for the server in order to make ssr works

not sure how this can work..

**Comment by damassi** [comment]
We're running into this issue as well, and specifically with loadable components. Without the manifest info there's no way to boot up bundle split server-side-render ready components. 

We could disable SSR rendering during development, but then the app wouldn't be accurate. Ideally webpack would be able to support both use cases.

**Comment by meskill** [comment]
Hello, @vankop I've created a [repo](https://github.com/meskill/webpack-lazy-compilation) to show the actual problem and provide some details on how the SSR works in our case.

Also maybe @sokra as a webpack lord can provide any suggestion on how to combine SSR and lazyCompilation together

**Reply by SuperOleg39** [reply]
@sokra Hello! Maybe you can help with this problem?
Reproduction repo is perfect, and we really want to test this optimization...

**Reply by SuperOleg39** [reply]
Maybe @alexander-akait  or  @TheLarkInn can help?
Reproduction repo is complete and have a good description https://github.com/meskill/webpack-lazy-compilation
